# Project: [Analyzing Students' Mental Health in SQL](https://app.datacamp.com/learn/projects/analyzing_students_mental_health)
![Illustration of silhouetted heads](mentalhealth.jpg)

Does going to university in a different country affect your mental health? A Japanese international university surveyed its students in 2018 and published a study the following year that was approved by several ethical and regulatory boards.

The study found that international students have a higher risk of mental health difficulties than the general population, and that social connectedness (belonging to a social group) and acculturative stress (stress associated with joining a new culture) are predictive of depression.


Explore the `students` data using DuckDB to find out if you would come to a similar conclusion for international students and see if the length of stay is a contributing factor.

Here is a data description of the columns you may find helpful.

| Field Name    | Description                                      |
| ------------- | ------------------------------------------------ |
| `inter_dom`     | Types of students (international or domestic)   |
| `japanese_cate` | Japanese language proficiency                    |
| `english_cate`  | English language proficiency                     |
| `academic`      | Current academic level (undergraduate or graduate) |
| `age`           | Current age of student                           |
| `stay`          | Current length of stay in years                  |
| `todep`         | Total score of depression (PHQ-9 test)           |
| `tosc`          | Total score of social connectedness (SCS test)   |
| `toas`          | Total score of acculturative stress (ASISS test) |

### Setup our environment and load in our data

JupySQL reference information:

- [Options](https://jupysql.ploomber.io/en/latest/api/configuration.html#options) - for the config SqlMagic settings

In [1]:
%load_ext sql
%sql duckdb:// --alias students
%config SqlMagic.autopandas = True
%config SqlMagic.displaycon = False

Connecting to 'students'

In [2]:
%%sql
CREATE TABLE students AS FROM 'data/students.csv'

Empty DataFrame
Columns: [Success]
Index: []

We can use the JupySQL [%sqlcmd profile](https://jupysql.ploomber.io/en/latest/api/magic-profile.html#sqlcmd-profile) to obtain summary statistics of the table. **Note:** the count used in the profile command excludes NULLS.

In [3]:
%sqlcmd profile --table students

+--------+-----------+--------+--------+----------+---------+----------+--------+-----------+----------+---------------+---------+--------------+----------+----------+---------+-----+---------+---------+--------+---------+---------+---------+---------+--------+--------+--------+----------+---------+---------+---------+---------+----------+---------+--------+--------+--------+--------+--------+----------+------------+------------+------------+-------------+-----------------+----------+-----------+-------------+----------+-----------+-------------+
|        | inter_dom | region | gender | academic |   age   | age_cate |  stay  | stay_cate | japanese | japanese_cate | english | english_cate | intimate | religion | suicide | dep | deptype |  todep  | depsev |   tosc  |   apd   |  ahome  |   aph   | afear  |  acs   | aguilt | amiscell |   toas  | partner | friends | parents | relative | profess | phone  | doctor |  reli  | alone  | others | internet | partner_bi | friends_bi | parents_bi | relative_bi | professional_bi | phone_bi | doctor_bi | religion_bi | alone_bi | others_bi | internet_bi |
+--------+-----------+--------+--------+----------+---------+----------+--------+-----------+----------+---------------+---------+--------------+----------+----------+---------+-----+---------+---------+--------+---------+---------+---------+---------+--------+--------+--------+----------+---------+---------+---------+---------+----------+---------+--------+--------+--------+--------+--------+----------+------------+------------+------------+-------------+-----------------+----------+-----------+-------------+----------+-----------+-------------+
| count  |    268    |  268   |  268   |   268    |   268   |   268    |  268   |    268    |   268    |      268      |   268   |     268      |   260    |   268    |   268   | 270 |   271   |   268   |  273   |   268   |   268   |   268   |   268   |  268   |  268   |  268   |   268    |   268   |   268   |   268   |   268   |   268    |   268   |  268   |  268   |  268   |  268   |  268   |   242    |    283     |    283     |    272     |     272     |       272       |   272    |    272    |     272     |   272    |    272    |     272     |
| unique |     2     |   5    |   2    |    2     |    14   |    5     |   9    |     3     |    5     |       3       |    5    |      3       |    2     |    2     |    2    |  4  |    6    |    26   |   10   |    38   |    27   |    15   |    18   |   14   |   11   |   9    |    32    |    80   |    7    |    7    |    7    |    7     |    7    |   7    |   7    |   7    |   7    |   7    |    7     |     14     |     14     |     4      |      4      |        4        |    4     |     4     |      4      |    4     |     4     |      4      |
|  top   |   Inter   |  SEA   | Female |  Under   |   nan   |   nan    |  nan   |   Medium  |   nan    |      Low      |   nan   |     High     |    No    |    No    |    No   |  No |    No   |   nan   |  Mild  |   nan   |   nan   |   nan   |   nan   |  nan   |  nan   |  nan   |   nan    |   nan   |   nan   |   nan   |   nan   |   nan    |   nan   |  nan   |  nan   |  nan   |  nan   |  nan   |    1     |    Yes     |     No     |    Yes     |      No     |        No       |    No    |     No    |      No     |    No    |     No    |      No     |
|  freq  |    201    |  122   |  170   |   247    |   nan   |   nan    |  nan   |    121    |   nan    |       92      |   nan   |     166      |   157    |   177    |   207   | 172 |   172   |   nan   |  107   |   nan   |   nan   |   nan   |   nan   |  nan   |  nan   |  nan   |   nan    |   nan   |   nan   |   nan   |   nan   |   nan    |   nan   |  nan   |  nan   |  nan   |  nan   |  nan   |    68    |    145     |    140     |    137     |     202     |       207       |   238    |    222    |     249     |   203    |    247    |     223     |
|  mean  |    nan    |  nan   |  nan   |   nan    | 20.8731 |  2.5597  | 2.1493 |    nan    |  3.0970  |      nan      |  3.6493 

DuckDB has some similiar functionality with the [SUMMARIZE](https://duckdb.org/docs/guides/meta/summarize) command which also includes statistics on the number of missing records (nulls). 


In [4]:
%sql SUMMARIZE students;

column_name column_type      min    max approx_unique  \
0         inter_dom     VARCHAR      Dom  Inter             2   
1            region     VARCHAR       EA    SEA             5   
2            gender     VARCHAR   Female   Male             2   
3          academic     VARCHAR     Grad  Under             2   
4               age      BIGINT       17     31            14   
5          age_cate      BIGINT        1      5             5   
6              stay      BIGINT        1     10             9   
7         stay_cate     VARCHAR     Long  Short             3   
8          japanese      BIGINT        1      5             5   
9     japanese_cate     VARCHAR  Average    Low             3   
10          english      BIGINT        1      5             5   
11     english_cate     VARCHAR  Average    Low             3   
12         intimate     VARCHAR       No    Yes             2   
13         religion     VARCHAR       No    Yes             2   
14          suicide     VARCHAR       No    Yes             2   
15              dep     VARCHAR      172    Yes             4   
16          deptype     VARCHAR      172  Other             6   
17            todep      BIGINT        0     25            26   
18           depsev     VARCHAR      107    Sev            10   
19             tosc      BIGINT        8     48            38   
20              apd      BIGINT        8     39            27   
21            ahome      BIGINT        4     20            15   
22              aph      BIGINT        5     25            18   
23            afear      BIGINT        4     17            14   
24              acs      BIGINT        3     13            11   
25           aguilt      BIGINT        2     10             9   
26         amiscell      BIGINT       10     47            32   
27             toas      BIGINT       36    145            80   
28          partner      BIGINT        1      7             7   
29          friends      BIGINT        1      7             7   
30          parents      BIGINT        1      7             7   
31         relative      BIGINT        1      7             7   
32          profess      BIGINT        1      7             7   
33            phone      BIGINT        1      7             7   
34           doctor      BIGINT        1      7             7   
35             reli      BIGINT        1      7             7   
36            alone      BIGINT        1      7             7   
37           others      BIGINT        1      7             7   
38         internet      BIGINT        1      7             7   
39       partner_bi     VARCHAR      123    Yes            14   
40       friends_bi     VARCHAR      123    Yes            14   
41       parents_bi     VARCHAR      131    Yes             4   
42      relative_bi     VARCHAR      202    Yes             4   
43  professional_bi     VARCHAR      207    Yes             4   
44         phone_bi     VARCHAR      238    Yes             4   
45        doctor_bi     VARCHAR      222    Yes             4   
46      religion_bi     VARCHAR       19    Yes             4   
47         alone_bi     VARCHAR      203    Yes             4   
48        others_bi     VARCHAR       21    Yes             4   
49      internet_bi     VARCHAR      223    Yes             4   

                   avg                 std   q25   q50   q75  count  \
0                 None                None  None  None  None    286   
1                 None                None  None  None  None    286   
2                 None                None  None  None  None    286   
3                 None                None  None  None  None    286   
4    20.87313432835821  2.7652786783892576    19    20    22    286   
5   2.5597014925373136  1.0706237044601954     2     2     3    286   
6   2.1492537313432836   1.326804969095303     1     2     3    286   
7                 None                None  None  None  None    286   
8   3.0970149253731343   1.306104382870079     2     

Based on the output we see that we have:
- 286 rows and 50 columns of data
- between 1 and 15 % of columns have some missing data
- looking at the original csv file we see that the last 18 lines of the file have mainly missing data which suggests the file might have been altered or corrupted.

### Start by counting all of the records in the data, then all records per student type to see how the records are categorized and scored.

We can use [FROM first](https://duckdb.org/2023/08/23/even-friendlier-sql.html#from-first-in-select-statements) with DuckDB because that's the way it should have always been! ;)



In [5]:
%%sql
-- count the number of record for each type of student
FROM students
SELECT 
	COUNT(*) AS total_records,
	COUNT(inter_dom) AS inter_dom_count,
	COUNT(*) FILTER (WHERE inter_dom == 'Inter') AS inter_count,
	COUNT(*) FILTER (WHERE inter_dom == 'Dom') AS dom_count,
	COUNT(*) FILTER (WHERE inter_dom IS NULL) AS missing_inter_dom_count;

total_records  inter_dom_count  inter_count  dom_count  \
0            286              268          201         67   

   missing_inter_dom_count  
0                       18

Another way is to use [histogram](https://duckdb.org/docs/sql/aggregates#general-aggregate-functions) to get a key-value pairing of our data.

In [6]:
%%sql
FROM students
SELECT histogram(inter_dom);

histogram(inter_dom)
0  {'key': ['Dom', 'Inter'], 'value': [67, 201]}

### Filter the data to see how it differs between the student types.

In [7]:
%%sql 
-- Review the data of international students
SUMMARIZE FROM students WHERE inter_dom = 'Inter';

column_name column_type      min    max approx_unique  \
0         inter_dom     VARCHAR    Inter  Inter             1   
1            region     VARCHAR       EA    SEA             5   
2            gender     VARCHAR   Female   Male             2   
3          academic     VARCHAR     Grad  Under             2   
4               age      BIGINT       17     31            14   
5          age_cate      BIGINT        1      5             5   
6              stay      BIGINT        1     10             9   
7         stay_cate     VARCHAR     Long  Short             3   
8          japanese      BIGINT        1      5             5   
9     japanese_cate     VARCHAR  Average    Low             3   
10          english      BIGINT        1      5             5   
11     english_cate     VARCHAR  Average    Low             3   
12         intimate     VARCHAR       No    Yes             2   
13         religion     VARCHAR       No    Yes             2   
14          suicide     VARCHAR       No    Yes             2   
15              dep     VARCHAR       No    Yes             2   
16          deptype     VARCHAR    Major  Other             3   
17            todep      BIGINT        0     25            23   
18           depsev     VARCHAR     Mild    Sev             5   
19             tosc      BIGINT       11     48            35   
20              apd      BIGINT        8     39            26   
21            ahome      BIGINT        4     20            15   
22              aph      BIGINT        5     25            18   
23            afear      BIGINT        4     17            14   
24              acs      BIGINT        3     13            11   
25           aguilt      BIGINT        2     10             9   
26         amiscell      BIGINT       10     47            32   
27             toas      BIGINT       36    145            77   
28          partner      BIGINT        1      7             7   
29          friends      BIGINT        1      7             7   
30          parents      BIGINT        1      7             7   
31         relative      BIGINT        1      7             7   
32          profess      BIGINT        1      7             7   
33            phone      BIGINT        1      7             7   
34           doctor      BIGINT        1      7             7   
35             reli      BIGINT        1      7             7   
36            alone      BIGINT        1      7             7   
37           others      BIGINT        1      7             7   
38         internet      BIGINT        1      7             7   
39       partner_bi     VARCHAR       No    Yes             2   
40       friends_bi     VARCHAR       No    Yes             2   
41       parents_bi     VARCHAR       No    Yes             2   
42      relative_bi     VARCHAR       No    Yes             2   
43  professional_bi     VARCHAR       No    Yes             2   
44         phone_bi     VARCHAR       No    Yes             2   
45        doctor_bi     VARCHAR       No    Yes             2   
46      religion_bi     VARCHAR       No    Yes             2   
47         alone_bi     VARCHAR       No    Yes             2   
48        others_bi     VARCHAR       No    Yes             2   
49      internet_bi     VARCHAR       No    Yes             2   

                   avg                 std   q25   q50   q75  count  \
0                 None                None  None  None  None    201   
1                 None                None  None  None  None    201   
2                 None                None  None  None  None    201   
3                 None                None  None  None  None    201   
4   21.029850746268657   3.033002551534032    19    20    22    201   
5    2.601990049751244  1.1272958883542945     2     2     3    201   
6   2.0646766169154227   1.375062187648434     1     2     3    201   
7                 None                None  None  None  None    201   
8   2.5223880597014925  0.9114528340274755     2     

In [8]:
%%sql
-- Review the data of domestic students
SUMMARIZE FROM students WHERE inter_dom = 'Dom';

column_name column_type      min    max approx_unique  \
0         inter_dom     VARCHAR      Dom    Dom             1   
1            region     VARCHAR      JAP    JAP             1   
2            gender     VARCHAR   Female   Male             2   
3          academic     VARCHAR     Grad  Under             2   
4               age      BIGINT       18     27             8   
5          age_cate      BIGINT        1      5             5   
6              stay      BIGINT        1      5             5   
7         stay_cate     VARCHAR     Long  Short             3   
8          japanese      BIGINT        2      5             4   
9     japanese_cate     VARCHAR  Average    Low             3   
10          english      BIGINT        1      5             5   
11     english_cate     VARCHAR  Average    Low             3   
12         intimate     VARCHAR       No    Yes             2   
13         religion     VARCHAR       No    Yes             2   
14          suicide     VARCHAR       No    Yes             2   
15              dep     VARCHAR       No    Yes             2   
16          deptype     VARCHAR    Major  Other             3   
17            todep      BIGINT        0     23            22   
18           depsev     VARCHAR     Mild    Sev             5   
19             tosc      BIGINT        8     48            26   
20              apd      BIGINT        8     28            18   
21            ahome      BIGINT        4     20            14   
22              aph      BIGINT        5     20            13   
23            afear      BIGINT        4     15            11   
24              acs      BIGINT        3     10             8   
25           aguilt      BIGINT        2      8             7   
26         amiscell      BIGINT       10     33            22   
27             toas      BIGINT       36    112            36   
28          partner      BIGINT        1      7             7   
29          friends      BIGINT        1      7             7   
30          parents      BIGINT        1      7             7   
31         relative      BIGINT        1      7             7   
32          profess      BIGINT        1      7             7   
33            phone      BIGINT        1      7             7   
34           doctor      BIGINT        1      7             7   
35             reli      BIGINT        1      6             6   
36            alone      BIGINT        1      7             7   
37           others      BIGINT        1      5             5   
38         internet      BIGINT        1      7             7   
39       partner_bi     VARCHAR       No    Yes             2   
40       friends_bi     VARCHAR       No    Yes             2   
41       parents_bi     VARCHAR       No    Yes             2   
42      relative_bi     VARCHAR       No    Yes             2   
43  professional_bi     VARCHAR       No    Yes             2   
44         phone_bi     VARCHAR       No    Yes             2   
45        doctor_bi     VARCHAR       No    Yes             2   
46      religion_bi     VARCHAR       No    Yes             2   
47         alone_bi     VARCHAR       No    Yes             2   
48        others_bi     VARCHAR       No    Yes             2   
49      internet_bi     VARCHAR       No    Yes             2   

                   avg                 std   q25   q50   q75  count  \
0                 None                None  None  None  None     67   
1                 None                None  None  None  None     67   
2                 None                None  None  None  None     67   
3                 None                None  None  None  None     67   
4   20.402985074626866   1.661139637030328    19    20    21     67   
5   2.4328358208955225  0.8742768543724359     2     2     3     67   
6   2.4029850746268657  1.1422956881507507     1     3     3     67   
7                 None                None  None  None  None     67   
8     4.82089552238806   0.601144694282912     5     

### Find the summary statistics of the diagnostic tests for all students using aggregate functions, rounding the test scores to two decimal places, remembering to use aliases.
The diagnotistic tests are under columns:
- `todep`
- `tosc`
- `toas`

In [9]:
%%sql
-- Domestic student test scores
FROM students
SELECT 
	stay, 
	ROUND(AVG(todep), 2) AS average_phq, -- PHQ test
	ROUND(AVG(tosc), 2) AS average_scs, -- SCS test
	ROUND(AVG(toas), 2) AS average_as -- ASISS test
WHERE inter_dom = 'Dom'
GROUP BY stay
ORDER BY stay DESC;	

stay  average_phq  average_scs  average_as
0     5        11.50        34.00       88.00
1     4         7.00        36.67       64.78
2     3         8.43        39.09       58.04
3     2         9.46        37.08       66.46
4     1         8.70        37.15       62.60

Your final query:

- See if length of stay impacts the average diagnostic scores rounded to two decimal places for international students, and order the results by descending order of the length of stay.
- Return a table with nine observations and five columns: `stay`, `count_int`, `average_phq`, `average_scs`, and `average_as`, **in that order**. The average columns should contain the average of the `todep`, `tosc`, and `toas` columns for each length of stay, **rounded to two decimal places**. The `count_int` column should be the number of international students for each length of stay. Sort the results by the length of stay in **descending order**.

In [10]:
%%sql
-- International student test scores
FROM students
SELECT 
	stay, 
	COUNT(*) AS count_int,
	ROUND(AVG(todep), 2) AS average_phq, -- PHQ test
	ROUND(AVG(tosc), 2) AS average_scs, -- SCS test
	ROUND(AVG(toas), 2) AS average_as -- ASISS test
WHERE inter_dom = 'Inter'
GROUP BY stay
ORDER BY stay DESC;	

stay  count_int  average_phq  average_scs  average_as
0    10          1        13.00        32.00       50.00
1     8          1        10.00        44.00       65.00
2     7          1         4.00        48.00       45.00
3     6          3         6.00        38.00       58.67
4     5          1         0.00        34.00       91.00
5     4         14         8.57        33.93       87.71
6     3         46         9.09        37.13       78.00
7     2         39         8.28        37.08       77.67
8     1         95         7.48        38.11       72.80

It seems that our data is potentially missing records from international students that have stayed 5 years for their PHQ test scores. 

Assuming that a higher score is meant to show a higher level of depression it would appear there might be some level of correlation between a higher length of stay and a higher risk of depression. 

In [30]:
%%sql
SELECT 
    CORR(todep, stay) AS phq_stay_corr,
    CORR(tosc, stay) AS scs_stay_corr,
    CORR(toas, stay) AS asiss_stay_corr
FROM students;

phq_stay_corr  scs_stay_corr  asiss_stay_corr
0       0.051581      -0.037587         0.000221